core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -350,7 +328,261 @@ public void run() {
     }
   }
 
-  static class NoopClientStream implements ClientStream {
+  /**
+   * A stream that queues requests before the transport is available, and delegates to a real stream
+   * implementation when the transport is available.
+   *
+   * <p>{@code ClientStream} itself doesn't require thread-safety. However, the state of {@code
+   * DelayedStream} may be internally altered by different threads, thus internal synchronization is
+   * necessary.
+   */
+  private class DelayedStream implements ClientStream {
+    final Metadata headers;
+    final ClientStreamListener listener;
+    final ListenableFuture<ClientTransport> transportFuture;
+
+    // volatile to be readable without synchronization in the fast path
+    @GuardedBy("this")
```

Comment:
GuardedBy means it is needed for both reads and writes. Remove the GuardedBy and just use documentation.